modules/jdbc/src/main/java/org/testcontainers/containers/JdbcDatabaseContainer.java

```diff
@@ -127,16 +127,17 @@ public Driver getJdbcDriverInstance() {
     /**
      * Creates a connection to the underlying containerized database instance.
      *
-     * @param queryString any special query string parameters that should be appended to the JDBC connection URL. The
-     *                    '?' character must be included
+     * @param queryString
+     *          query string parameters that should be appended to the JDBC connection URL.
+     *          The '?' character must be included
      * @return a Connection
      * @throws SQLException if there is a repeated failure to create the connection
      */
     public Connection createConnection(String queryString) throws SQLException {
         final Properties info = new Properties();
         info.put("user", this.getUsername());
         info.put("password", this.getPassword());
-        final String url = appendDisableSslConfig(this.getJdbcUrl() + queryString);
+        final String url = constructUrlForConnection(queryString);
```

Comment:
I wonder if the real problem lies in line 145. I'm not a JDBC expert, but shouldn't we provide those flags in the `Properties` object, like `info.setProperty("useSSL", "false");`? AFAIK this would make this condfigs database driver implementation independent, at least regarding the way, the url is constructed.

I'm totally fine with this fix ATM, but I think we should refactor this part in a follow up, ideally once we've changed to the monorepo, so we can easily test for regressions.

Reply:
I got inspired by your comment, but after testing this with Hive-jdbc i got sad.
I'm not gonna force you into supporting each and every corner use case. Since it's implementation specific. Ergo I should probably submit a ticket to hive-jdbc developers.

But there is no guarantee that all information in info.properties will have affect. I assume that the reason is that most end-user applications accepts a String to make the connection, usually with user/pass fields, but not a key-value form for setting the info properties.

So the best bet is that parameters in ConnectionString will work and that info properties might work.

And just to show some of the horror with hive-jdbc connection string.
Hive JDBC URL: jdbc:hive2://<host>:<port>/dbName;sess_var_list?hive_conf_list#hive_var_list 

schema:host:port/path;matrix?query#fragment